Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -682,15 +682,21 @@ class CombinatorialKalmanFilter {
       navigationOptions.startSurface = &currentState.referenceSurface();
       navigationOptions.targetSurface = nullptr;
       state.navigation = navigator.makeState(navigationOptions);
-      navigator.initialize(state, stepper);
+      navigator.initialize(state.navigation, stepper.position(state.stepping),
+                           stepper.direction(state.stepping),
+                           state.options.direction);
 
       // No Kalman filtering for the starting surface, but still need
       // to consider the material effects here
       materialInteractor(navigator.currentSurface(state.navigation), state,
                          stepper, navigator, MaterialUpdateStage::PostUpdate);
 
+      // Set path limit based on loop protection
       detail::setupLoopProtection(state, stepper, result.pathLimitReached, true,
                                   logger());
+
+      // Set path limit based on target surface
+      targetReached.checkAbort(state, stepper, navigator, logger());
```

Comment:
Is this new behavior or was this done some other way before?

Reply:
This was missing before and I put it in because I thought it became a problem. Turned out it was just the invalidation of the current target which was the problem. I would still like to get this is but we could factor this out